### PR TITLE
[RV64_DYNAREC] Implementation of some CRC32 instructions

### PR DESCRIPTION
**Pull request by kaixindeken**

### CODE REVIEW

**Comment by ksco** [comment]
How is this tested? Is it covered by our unit tests or it’s used by some real-world programs?

File: src/dynarec/rv64/dynarec_rv64_66f20f.c [hunk 1]
```diff
@@ -51,6 +51,31 @@ uintptr_t dynarec64_66F20F(dynarec_rv64_t* dyn, uintptr_t addr, uintptr_t ip, in
...
+                    GETGD;
+                    GETEW(x1, 0);
+                    MOV32w(x2, 0x82f63b78);
+                    for(int j=0; j<2; ++j) {
+                        SRLI(x5, ed, 8*j);
+                        ANDI(x3, x5, 0xFF);
+                        XOR(gd, gd, x3);
+                        for (int i = 0; i < 8; i++) {
+                            SRLI((i&1)?gd:x4, (i&1)?x4:gd, 1);
+                            BEXTI(x6, (i&1)?x4:gd, 0);
```
Comment (ksco) [inline src/dynarec/rv64/dynarec_rv64_66f20f.c:69]: Prefer `ANDI(x6, (i&1)?x4:gd, 1);`

File: src/dynarec/rv64/dynarec_rv64_f20f.c [hunk 2]
```diff
@@ -147,6 +147,38 @@ uintptr_t dynarec64_F20F(dynarec_rv64_t* dyn, uintptr_t addr, uintptr_t ip, int
...
+                case 0xF0:  // CRC32 Gd, Eb
+                    INST_NAME("CRC32 Gd, Eb");
+                    nextop = F8;
+                    GETEB(x1, 0);
+                    GETGD;
+                    XOR(gd, gd, ed);
+                    MOV32w(x2, 0x82f63b78);
+                    for (int i = 0; i < 8; i++) {
+                        SRLI((i&1)?gd:x4, (i&1)?x4:gd, 1);
+                        BEXTI(x6, (i&1)?x4:gd, 0);
```
Comment (ksco) [inline src/dynarec/rv64/dynarec_rv64_f20f.c:159]: Prefer `ANDI`

File: src/dynarec/rv64/dynarec_rv64_f20f.c [hunk 3]
```diff
@@ -147,6 +147,38 @@ uintptr_t dynarec64_F20F(dynarec_rv64_t* dyn, uintptr_t addr, uintptr_t ip, int
...
+                    GETGD;
+                    GETED(0);
+                    MOV32w(x2, 0x82f63b78);
+                    for(int j=0; j<4*(rex.w+1); ++j) {
+                        SRLI(x5, ed, 8*j);
+                        ANDI(x3, x5, 0xFF);
+                        XOR(gd, gd, x3);
+                        for (int i = 0; i < 8; i++) {
+                            SRLI((i&1)?gd:x4, (i&1)?x4:gd, 1);
+                            BEXTI(x6, (i&1)?x4:gd, 0);
```
Comment (ksco) [inline src/dynarec/rv64/dynarec_rv64_f20f.c:176]: Prefer `ANDI`

**Comment by kaixindeken** [comment]
These implementation is needed by the newest wechat-beta for web connection. I'll correct it later. Thanks.

File: src/dynarec/rv64/dynarec_rv64_f20f.c (hunk 3 above)
Comment (kaixindeken) [reply src/dynarec/rv64/dynarec_rv64_f20f.c:176]: > Prefer `ANDI`

 I'll correct it later. Thanks.

**Review by ksco — APPROVED** [review]

**Pull request merged**